src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -178,6 +178,19 @@ public int getTotalMessageCount(){
        foodnames.add((String) entity.getProperty("Name"));
     }
     return foodnames;
+  }
+
+  public double getCO2forOneUnitFood(String FoodName){
+    Map<String,Double> namesToCO2 = new HashMap<>();
+    Query query = new Query("FoodItem");
+    PreparedQuery results = datastore.prepare(query);
+    for(Entity entity : results.asIterable()) {
+       namesToCO2.put((String) entity.getProperty("Name"), (double)entity.getProperty("CO2"));
+    }
+    if(namesToCO2.containsKey(FoodName)){
+      return namesToCO2.get(FoodName);
+    }
```

Comment:
As an efficiency tip (meaning you don't need to do this), it's probably faster (you never know until you measure it!) to compare the food name against the results directly while iterating over them since hashing the food names and inserting into the Map is likely as expensive as the repeated string comparisons.